### ISSUE TITLE
carbon v10, accessibility issues with TableToolBar search

### ISSUE THREAD

**Issue by sakthi-nathiarasan** [carbon-design-system/carbon#5630]
## Environment

> Operating system
Linux -RHEL

> Browser
Chrome

**Comment by sakthi-nathiarasan** [comment]
Hi, with latest carbon packages,
carbon-components=10.10.0,
carbon-components-react=7.10.0,
There are few accesibility issues with tabletoolbarsearch

-If we use Tabs (via keyboard) to access tabletoolbarsearch, focus is not getting clear (no blue highlight unlike other components)
-After getting focused via Tab, we cannot type anything in searchbox
-We have to double click to enter search input

You can reproduce these issues in carbon website itself,

https://react.carbondesignsystem.com/?path=/story/datatable--with-toolbar

These are high priority accesibility bugs

@joshblack @amuqeetkhan @asudoh
